src/android/com/adobe/phonegap/push/PushPlugin.java

```diff
@@ -179,6 +181,20 @@ public void run() {
                     }
                 }
             });
+        } else if (SET_APPLICATION_ICON_BADGE_NUMBER.equals(action)) {
```

Comment:
Okay so your patch (https://github.com/phonegap/phonegap-plugin-push/compare/master...silo-co:master) handles the updating of the push in the background, and mine in foreground for `setApplicationIconBadgeNumber` right?

So we can combine both methods, right? Can you submit a PR to my branch to add that?